### PR TITLE
Update UI test plan

### LINKED ISSUE #186: Perform manual testing of Teleport V5
## Main
For main, test with admin role that has access to all resources.

#### Top Nav
- [x] Verify that cluster selector displays all (root + leaf) clusters    
- [x] Verify that user name is displayed
- [x] Verify that user menu shows logout, help&support, and account settings

#### Side Nav
- [x] Verify that each item has an icon
- [x] Verify that Collapse/Expand works and collapsed has icon `>`, and expand has icon `v`
- [x] Verify that it automatically expands and highlights the item on page refresh

#### Servers aka Nodes
- [x] Verify that "Servers" table shows all joined nodes
- [x] Verify that "Connect" button shows a list of available logins
- [x] Verify that "Hostname", "Address" and "Labels" columns show the current values
- [x] Verify that "Search" by hostname, address, labels works
- [x] Verify that terminal opens when clicking on one of the available logins
- [x] Verify that clicking on `Add Server` button renders dialogue set to `Automatically` view
  - [x] Verify clicking on `Regenerate Script` regenerates token value in the bash command
  - [x] Verify using the bash command successfully adds the server (refresh server list)
  - [x] Verify that clicking on `Manually` tab renders manual steps
  - [x] Verify that clicking back to `Automatically` tab renders bash command

#### Applications
- [x] Verify that all apps registered is shown
- [x] Verify that clicking on the app icon takes you to another tab
- [x] Verify that clicking on `Add Application` button renders dialogue
  - [x] Verify input validation (prevent empty value and invalid url)
  - [x] Verify after input and clicking on `Generate Script`, bash command is rendered
  - [x] Verify clicking on `Regenerate` button regenerates token value in bash command
  - [x] Verify using the bash command successfully adds the application (refresh app list)

#### Active Sessions
- [x] Verify that "empty" state is handled
- [x] Verify that it displays the session when session is active
- [x] Verify that "Description", "Session ID", "Users", "Nodes" and "Duration" columns show correct values
- [x] Verify that "OPTIONS" button allows to join a session

#### Audit log
- [x] Verify that time range button is shown and works
- [x] Verify clicking on `Session Ended` event icon, takes user to session player
- [x] Verify event detail dialogue renders when clicking on events `details` button
- [x] Verify searching by type, description, created works

#### Users
- [x] Verify that users are shown
- [x] Verify that creating a new user works
- [x] Verify that editing user roles works
- [x] Verify that removing a user works
- [x] Verify resetting a users password works
- [x] Verify search by username, roles, and type works

#### Auth Connectors
- [ ] Verify that creating OIDC/SAML/GITHUB connectors works
- [ ] Verify that editing  OIDC/SAML/GITHUB connectors works
- [ ] Verify that error is shown when saving an invalid YAML
- [ ] Verify that correct hint text is shown on the right side

#### Auth Connectors Card Icons
- [x] Verify that GITHUB card has github icon
- [x] Verify that SAML card has SAML icon
- [x] Verify that OIDC card has OIDC icon
- [x] Verify when there are no connectors, empty state renders

#### Roles
- [x] Verify that roles are shown
- [x] Verify that "Create New Role" dialog works
- [x] Verify that deleting and editing works
- [x] Verify that error is shown when saving an invalid YAML
- [ ] Verify that correct hint text is shown on the right side

#### Managed Clusters
- [x] Verify that it displays a list of clusters (root + leaf)
- [x] Verify that every menu item works: nodes, apps, audit events, session recordings.

#### Help&Support
- [x] Verify that all URLs work and correct (no 404)

#### Strategy Reason
Create the following role:
```
kind: role
metadata:
  name: restrict
spec:
  allow:
    request:
      roles:
      - <some other role to assign user after approval>
  deny: {}
  options:
    cert_format: standard
    enhanced_recording:
    - command
    - network
    forward_agent: false
    max_session_ttl: 8h0m0s
    port_forwarding: true
    request_access: reason
    request_prompt: <some custom prompt to show in reason dialogue>
version: v3
```
- [x] Verify after login, reason dialogue is rendered with prompt set to `request_prompt` setting
- [x] Verify after clicking `send request`, pending dialogue renders
- [x] Verify after `tctl requests approve <request-id>`, dashboard is rendered
- [x] Verify the correct role was assigned

#### Strategy Always
With the previous role you created from `Strategy Reason`, change `request_access` to `always`:
- [x] Verify after login, pending dialogue is rendered
- [x] Verify after `tctl requests approve <request-id>`, dashboard is rendered
- [x] Verify after login, `tctl requests deny <request-id>`, access denied dialogue is rendered

#### Strategy Optional
With the previous role you created from `Strategy Reason`, change `request_access` to `optional`:
- [x] Verify after login, dashboard is rendered

## Account
- [x] Verify that Account screen is accessibly from the user menu for local users.
- [ ] Verify that changing a local password works (OTP, U2F)

## Terminal
- [x] Verify that top nav has a user menu (Main and Logout)
- [x] Verify that switching between tabs works on alt+[1...9]

#### Node List Tab
- [x] Verify that Cluster selector works (URL should change too)
- [x] Verify that Quick launcher input works
- [x] Verify that Quick launcher input handles input errors
- [x] Verify that "Connect" button shows a list of available logins
- [x] Verify that "Hostname", "Address" and "Labels" columns show the current values
- [x] Verify that "Search" by hostname, address, labels work
- [x] Verify that new tab is created when starting a session

#### Session Tab
- [x] Verify that session and browser tabs both show the title with login and node name
- [x] Verify that terminal resize works
    - Install midnight commander on the node you ssh into: `$ sudo apt-get install mc`
    - Run the program: `$ mc`
    - Resize the terminal to see if panels resize with it
- [x] Verify that session tab shows/updates number of participants when a new user joins the session
- [x] Verify that tab automatically closes on "$ exit" command
- [x] Verify that SCP Upload works
- [x] Verify that SCP Upload handles invalid paths and network errors
- [x] Verify that SCP Download works
- [x] Verify that SCP Download handles invalid paths and network errors

## Session Player
- [x] Verify that it can replay a session
- [x] Verify that when playing, scroller auto scrolls to bottom most content
- [x] Verify when resizing player to a small screen, scroller appears and is working
- [x] Verify that error message is displayed (enter a invalid SID in the URL)

## Invite Form
- [ ] Verify that input validation.
- [ ] Verify that invite works with 2FA disabled.
- [ ] Verify that invite works with OTP enabled.
- [ ] Verify that invite works with U2F enabled.
- [ ] Verify that error message is shown if an invite is expired/invalid.

## Login Form
- [ ] Verify that input validation.
- [ ] Verify that login works with 2FA disabled.
- [ ] Verify that login works with OTP enabled.
- [ ] Verify that login works with U2F enabled.
- [ ] Verify that login works for Github/SAML/OIDC.
- [ ] Verify that account is locked after several unsuccessful attempts.
- [ ] Verify that redirect to original URL works after successful login.

## RBAC
Create a role, with no allow rules defined:
```
kind: role
metadata:
  name: test
spec:
  allow:
    app_labels:
      '*': '*'
    logins:
    - root
    node_labels:
      '*': '*'
  deny: {}
  options:
    cert_format: standard
    enhanced_recording:
    - command
    - network
    forward_agent: false
    max_session_ttl: 8h0m0s
    port_forwarding: true
version: v3
```
- [x] Verify that a user has access only to: "Servers", "Applications", "Active Sessions" and "Manage Clusters"
- [x] Verify there is no `Add Server` button in Server view
- [x] Verify there is no `Add Application` button in Applications view
- [x] Verify only `Nodes` and `Apps` are listed under `options` button in `Manage Clusters`

Add the following resource under `spec.allow.rules` to enable read access to the audit log:
```
  - resources:
      - event
      verbs:
      - list
```
- [x] Verify that the `Audit Log` and `Session Recordings` is accessible
- [x] Verify that playing a recorded sessions is denied

Add the following resource to enable read access to recorded sessions
```
  - resources:
      - session
      verbs:
      - read
```
- [x] Verify that a user can re-play a session (session.end)

Add the following resource to enable read access to the roles

```
- resources:
      - role
      verbs:
      - list
      - read
```
- [x] Verify that a user can see the roles
- [x] Verify that a user cannot create/delete/update a role

Add the following resource to enable read access to the auth connectors

```
- resources:
      - auth_connector
      verbs:
      - list
      - read
```
- [x] Verify that a user can see the list of auth connectors.
- [x] Verify that a user cannot create/delete/update the connectors

Add the following resource to enable read access to users
```
  - resources:
      - user
      verbs:
      - list
      - read
```
- [x] Verify that a user can access the "Users" screen
- [x] Verify that a user cannot create/delete/update a user

Add the following resource to enable read access to trusted clusters
```
  - resources:
      - trusted_cluster
      verbs:
      - list
      - read
```
- [x] Verify that a user can access the "Trust" screen
- [x] Verify that a user cannot create/delete/update a trusted cluster.

### PR DESCRIPTION
**Pull request by kimlisa**
resolves https://github.com/gravitational/webapps/issues/186

#### Description
Update web ui test plan to fit with the newest features and new navigation

### CODE REVIEW

**Review by alex-kovoy — APPROVED** [review]

File: docs/testplan.md [hunk 1]
```diff
@@ -200,170 +200,228 @@ tsh --proxy=proxy.example.com --user=<username> --insecure ssh --cluster=foo.com
...
+- [ ] Verify that "Search" by hostname, address, labels works
+- [ ] Verify that terminal opens when clicking on one of the available logins
+- [ ] Verify that clicking on `Add Server` button renders dialogue set to `Automatically` view
+  - [ ] Verify clicking on `Regenerate Script` regenerates token value in the bash command
+  - [ ] Verify using the bash command successfully adds the server (refresh server list)
+  - [ ] Verify that clicking on `Manually` tab renders manual steps
+  - [ ] Verify that clicking back to `Automatically` tab renders bash command
+
+#### Applications
+- [ ] Verify that all apps registered is shown
```
Comment (webvictim) [inline docs/testplan.md:229]:
```suggestion
- [ ] Verify that all apps registered are shown
```

File: docs/testplan.md [hunk 2]
```diff
@@ -200,170 +200,228 @@ tsh --proxy=proxy.example.com --user=<username> --insecure ssh --cluster=foo.com
...
-- [ ] Verify that Play button opens a session player.
-- [ ] Verify that search works (it should do a search on all table cells).
-
-#### Auth Connectors.
-- [ ] Verify that creating OIDC/SAML/GITHUB connectors works.
-- [ ] Verify that editing  OIDC/SAML/GITHUB connectors works.
-- [ ] Verify that error is shown when saving an invalid YAML.
-- [ ] Verify that correct hint text is shown on the right side.
+- [ ] Verify that time range button is shown and works
+- [ ] Verify clicking on `Session Ended` event icon, takes user to session player
```
Comment (webvictim) [inline docs/testplan.md:245]:
```suggestion
- [ ] Verify that clicking on `Session Ended` event icon takes user to session player
```

File: docs/testplan.md [hunk 3]
```diff
@@ -200,170 +200,228 @@ tsh --proxy=proxy.example.com --user=<username> --insecure ssh --cluster=foo.com
...
+- [ ] Verify event detail dialogue renders when clicking on events `details` button
+- [ ] Verify searching by type, description, created works
+
+#### Users
+- [ ] Verify that users are shown
+- [ ] Verify that creating a new user works
+- [ ] Verify that editing user roles works
+- [ ] Verify that removing a user works
+- [ ] Verify resetting a users password works
```
Comment (webvictim) [inline docs/testplan.md:254]:
```suggestion
- [ ] Verify resetting a user's password works
```

File: docs/testplan.md [hunk 4]
```diff
@@ -200,170 +200,228 @@ tsh --proxy=proxy.example.com --user=<username> --insecure ssh --cluster=foo.com
...
   - resources:
       - event
       verbs:
       - list
 ```
-- [ ] Verify that the audit log is accessible via UI.
-- [ ] Verify that access to recorded sessions is denied (by opening a session player).
-
+- [ ] Verify that the `Audit Log` and `Session Recordings` is accessible
+- [ ] Verify that playing a recorded sessions is denied
```
Comment (webvictim) [inline docs/testplan.md:415]:
```suggestion
- [ ] Verify that playing a recorded session is denied
```

**Review by webvictim — APPROVED** [review]

**Comment by webvictim** [comment]
(Drone is failing due to trailing whitespace - https://drone.gravitational.io/gravitational/teleport/3337/2/2)

File: docs/testplan.md (hunk 4 above)
Comment (kimlisa) [reply docs/testplan.md]: revised

File: docs/testplan.md (hunk 3 above)
Comment (kimlisa) [reply docs/testplan.md]: revised

File: docs/testplan.md (hunk 2 above)
Comment (kimlisa) [reply docs/testplan.md]: revised

File: docs/testplan.md (hunk 1 above)
Comment (kimlisa) [reply docs/testplan.md]: revised

**Pull request merged**